src/Parser/DeathKnight/Blood/Modules/Features/BoneShieldStacksBySeconds.js

```diff
@@ -0,0 +1,84 @@
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+
+/* 
+  boneShieldStacksBySeconds() returns an array with the amount of stacks with an entry for each second
```

Comment:
I would suggest using the exact timestamps to sum time between buff stack changes instead of only having one entry per second. It would make the uptimes more accurate.